sbomify_action/enrichment.py

```diff
@@ -0,0 +1,590 @@
+            if isinstance(data, list) and len(data) > 0:
+                return data[0]
+            elif isinstance(data, dict):
+                return data
+            else:
+                logger.debug(f"No package data found for: {purl}")
+                return None
+        elif response.status_code == 404:
+            logger.debug(f"Package not found in ecosyste.ms: {purl}")
+            return None
+        else:
+            logger.warning(f"Failed to fetch metadata for {purl}: HTTP {response.status_code}")
+            return None
+
+    except requests.exceptions.Timeout:
+        logger.warning(f"Timeout fetching metadata for {purl}")
+        return None
+    except requests.exceptions.RequestException as e:
+        logger.warning(f"Error fetching metadata for {purl}: {e}")
+        return None
+    except Exception as e:
+        logger.error(f"Unexpected error fetching metadata for {purl}: {e}")
+        return None
+
+
+def _fetch_all_metadata_sequential(purls: List[str]) -> Dict[str, Optional[Dict[str, Any]]]:
+    """
+    Fetch metadata for all PURLs sequentially.
+
+    Args:
+        purls: List of package URLs
+
+    Returns:
+        Dictionary mapping PURL to metadata
+    """
+    metadata_map = {}
+
+    # Use a single session for sequential requests
+    with requests.Session() as session:
+        session.headers.update({"User-Agent": USER_AGENT})
+
+        # Fetch metadata sequentially
+        for purl in purls:
+            try:
+                metadata = _fetch_package_metadata(purl, session)
+                metadata_map[purl] = metadata
+            except Exception as e:
+                logger.error(f"Unexpected error fetching metadata for {purl}: {e}")
+                metadata_map[purl] = None
+
+    return metadata_map
+
+
+def _enrich_cyclonedx_component(
+    component: Dict[str, Any], metadata: Optional[Dict[str, Any]]
+) -> Tuple[Dict[str, Any], List[str]]:
+    """
+    Enrich a CycloneDX component with ecosyste.ms metadata using ONLY native CycloneDX fields.
+
+    Native fields used:
+    - description: Component description
+    - licenses: Array of license objects
+    - publisher: Publishing entity/primary maintainer
+    - externalReferences: URLs for website, vcs, distribution, issue-tracker, etc.
+
+    Args:
+        component: Original component dict
+        metadata: Metadata from ecosyste.ms API
+
+    Returns:
+        Tuple of (enriched component dict, list of added fields)
+    """
+    if not metadata:
+        return component, []
+
+    enriched = component.copy()
+    added_fields = []
+
+    # Add description (native CycloneDX field)
+    if "description" not in enriched or not enriched["description"]:
+        if metadata.get("description"):
+            enriched["description"] = metadata["description"]
+            added_fields.append("description")
+
+    # Add licenses (native CycloneDX field)
+    if metadata.get("licenses"):
```

Comment:
probably want to use `normalized_licenses` field here as its an array and also tries to turn license names into spdx identifiers